### PR TITLE
parse: fix IsolationOption function

### PR DESCRIPTION
**Pull request by q384566678**
1.parse: modify the verification of the isolation value

To verify the value of the isolation, verify that all strings are the same, not just the beginning of the string.
Avoid users mistakenly thinking that the wrong value entered is also a correct value.

```
➜  buildah git:(isolation-fix) ✗ buildah from --isolation chrootcc cc510acfcd70
fedora-working-container-7
```

After Change:
```
➜  buildah git:(isolation-fix) ✗ ./buildah from --isolation chrootcc cc510acfcd70
unrecognized isolation type "chrootcc"
ERRO[0000] exit status 1
```

2.parse: Modify the return value

Code optimization and modification of return values.

Signed-off-by: Zhou Hao <zhouhao@cn.fujitsu.com>

### CODE REVIEW

**Comment by vrothberg** [comment]
bot, retest this please

**Comment by vrothberg** [comment]
Thanks for the PR! Overall LGTM. I am not a fan of mixing types (i.e., `Isolation` but returning `0`) but that's not a blocker given we don't have a `IsolationNone` or something similar.

**Comment by TomSweeneyRedHat** [comment]
test failure seems to be a network error pulling alpine, going to retest.

**Comment by TomSweeneyRedHat** [comment]
bot, retest this please.

**Comment by TomSweeneyRedHat** [comment]
LGTM if we can get the test to turn green

File: pkg/parse/parse.go [hunk 1]
```diff
@@ -543,11 +543,11 @@ func NamespaceOptions(c *cli.Context) (namespaceOptions buildah.NamespaceOptions
 func defaultIsolation() (buildah.Isolation, error) {
 	isolation, isSet := os.LookupEnv("BUILDAH_ISOLATION")
 	if isSet {
-		if strings.HasPrefix(strings.ToLower(isolation), "oci") {
+		if strings.ToLower(isolation) == "oci" {
```
Comment (rhatdan) [inline pkg/parse/parse.go]: Can we do this as a switch statement?

Comment (q384566678) [reply pkg/parse/parse.go]: updated, PTAL.

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Pull request closed without merging**